src/raw_client.rs

```diff
@@ -472,15 +465,13 @@ impl<S: Read + Write> RawClient<S> {
                             );
 
                             // Remove ourselves from the "waiting map"
-                            let mut map = self.waiting_map.lock().unwrap();
+                            let mut map = self.waiting_map.lock()?;
                             map.remove(&resp_id);
 
                             // If the map is not empty, we select a random thread to become the
                             // new reader thread.
                             if let Some(sender) = map.values().next() {
-                                sender
-                                    .send(ChannelMessage::WakeUp)
-                                    .expect("Unable to WakeUp a different thread");
+                                sender.send(ChannelMessage::WakeUp)?;
```

Comment:
If we are unable to wake up the thread I guess we should try with all the others before giving up, otherwise all the other threads that are still alive would remain hanging forever waiting for their response.

Reply:
Agree, will update